src/oops/designPatterns/singleton/SQLConnectionSingleton.java

```diff
@@ -17,7 +17,7 @@ private SQLConnectionSingleton() {
 	
 	public static SQLConnectionSingleton getInstance() {
 		if(instance == null) {
-			instance = new SQLConnectionSingleton();
+			instance = SqlConnectionSingletonInstanceProvider.INSTANCE;
```

Comment:
Nice idea!

But then, why can't instance variable itself be eagerly initiated in static block?

Reply:
Then that would not be lazy loading and would be consuming additional resources if it was not required from the beginning. The inner static class is loaded only when `getInstance()` is called and thus also takes care of lazy loading and as a class loading is done only once, the instance is also created only once.

Reply:
It is eager loading in current version as well. On line 20 you are only assigning value of `SqlConnectionSingletonInstanceProvider.INSTANCE` to `instance`. However you have already created this instance eagerly on line 47

Reply:
Yes, the instance is eagerly created in `SqlConnectionSingletonInstanceProvider` at line 47, but the class `SqlConnectionSingletonInstanceProvider` will only be loaded when `getInstance()` method is called. This makes it lazy-loading.

Ref - https://www.journaldev.com/1377/java-singleton-design-pattern-best-practices-examples#bill-pugh-singleton

Reply:
Oh got it. Thanks! :)